src/processors/receive.mjs

```diff
@@ -0,0 +1,89 @@
+
+let bufferLength,
+    checksumFails,
+    decoderPtr,
+    framePtr,
+    samplesPtr,
+    samplesView
+
+function init( {
+    activeProfileKey,
+    bufferSize,
+    profiles,
+    sampleRate,
+} ) {
+    bufferLength = bufferSize
+
+    const profilePtr = Module.ccall(
+        'quiet_decoder_profile_str', 'pointer', [ 'array', 'array' ], [
+            Module.intArrayFromString( JSON.stringify( profiles ) ),
+            Module.intArrayFromString( activeProfileKey )
+        ]
+    )
+    
+    decoderPtr = Module.ccall(
+        'quiet_decoder_create', 'pointer',
+        ['pointer', 'number'],
+        [ profilePtr, sampleRate ]
+    )
+    
+    Module.ccall( 'free', null, [ 'pointer' ], [ profilePtr ] )
+    
+    framePtr = Module.ccall(
+        'malloc', 'pointer',
+        [ 'number' ],
+        [ bufferSize ]
+    )
+
+    samplesPtr = Module.ccall(
+        'malloc', 'pointer',
+        [ 'number' ],
+        [ bufferSize * 4 ]
+    )
+    
+    // since pointers are byte indexes in emscripten's fake heap, we need to
+    // convert to double word indexes by dividing by ( 32 / 8 )
+    const dwordIndex = samplesPtr / 4
+    samplesView = Module.HEAPF32.subarray( dwordIndex, dwordIndex + bufferSize )
+    
+    checksumFails = 0
+}
```

Comment:
Does  init enforce one receiver versus the multiple receivers the original quietjs had? Doesn't matter much for the jitsi use case.